src/G2/Equiv/EquivADT.hs

```diff
@@ -27,7 +27,10 @@ import Data.Maybe
 -- earlier DataCons in the list are farther out
 -- the first Int tag indicates which argument of the constructor this was
 -- the second one indicates the total number of arguments for that constructor
+-- TODO keep track of data constructors on the way
+-- the lambda pair at the end is necessarily the last thing
 data Obligation = Ob [(DataCon, Int, Int)] Expr Expr
+                | Lams [(DataCon, Int, Int)] Expr Expr
```

Comment:
Do we need this?  Why not just have `Ob`, and then check in state wrap if the Expr's are Lambda's or not?  (In state wrap, it looks like we count on the Expr's in a Lam being Lambdas anyway)